pkg/exporter/elasticsearch/factory.go

```diff
@@ -46,6 +46,10 @@ func (f Factory) CreateDefaultConfig() configmodels.Exporter {
 		bulkWorkers:       opts.GetPrimary().BulkWorkers,
 		bulkFlushInterval: opts.GetPrimary().BulkFlushInterval,
 		Version:           opts.GetPrimary().Version,
+		ExporterSettings: configmodels.ExporterSettings{
```

Comment:
@objectiser I will ping you directly here. This is the convention other factories follow 

https://github.com/open-telemetry/opentelemetry-collector/blob/master/receiver/jaegerreceiver/factory.go#L106 
https://github.com/open-telemetry/opentelemetry-collector/blob/master/processor/queuedprocessor/factory.go#L45

Then the actual name is overriden when config is loaded.